File: ascenso-backend/src/main/java/com/ais/ascensobackend/catalogo/application/services/interfaces/LibroBiblicoService.java
package com.ais.ascensobackend.catalogo.application.services.interfaces;

import com.ais.ascensobackend.catalogo.application.dtos.LibroBiblicoResumen;
import java.util.List;

public interface LibroBiblicoService {

    List<LibroBiblicoResumen> listar(Long grupoId);

    LibroBiblicoResumen obtener(Long id);

    LibroBiblicoResumen crear(Long grupoId, String titulo, Short ordenSugerido);

    LibroBiblicoResumen actualizar(Long id, String titulo, Short ordenSugerido);

    void activar(Long id);

    void desactivar(Long id);
}

File: ascenso-backoffice/src/services/catalogo/LibroBiblicoService.ts
import { apiClient } from '@/services/http/ApiClient'
import { API_ENDPOINTS } from '@/config/endpoints'
import type { GuardarLibroBiblicoRequest, LibroBiblico } from '@/types/catalogo'

class LibroBiblicoService {
  listar(grupoId?: number) {
    return apiClient.get<LibroBiblico[]>(API_ENDPOINTS.catalogo.librosBiblicos, {
      params: grupoId ? { grupoId } : undefined,
    })
  }

  crear(request: GuardarLibroBiblicoRequest) {
    return apiClient.post<LibroBiblico>(API_ENDPOINTS.catalogo.librosBiblicos, request)
  }

  actualizar(id: number, request: GuardarLibroBiblicoRequest) {
    return apiClient.put<LibroBiblico>(API_ENDPOINTS.catalogo.libroBiblicoPorId(id), request)
  }

  activar(id: number) {
    return apiClient.post<void>(API_ENDPOINTS.catalogo.libroBiblicoActivar(id))
  }

  desactivar(id: number) {
    return apiClient.post<void>(API_ENDPOINTS.catalogo.libroBiblicoDesactivar(id))
  }
}

export const libroBiblicoService = new LibroBiblicoService()
